### PR TITLE
Profile/38952/add update gender identity

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Description
Adds the ability to add and update a gender identity to the personal information page of profile.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/38952

## Testing done
Updated unit tests and added E2E tests

## Acceptance criteria
- [x] Gender identity can be updated on profile

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/mocks/personal-information/index.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
+/* eslint-disable camelcase */
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/mocks/personal-information/index.js:1]: Moved this mock into its own folder to adhere to the other mock's organizational structure

File: src/applications/personalization/profile/actions/personalInformation.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ export function createPersonalInfoUpdate({
       set(
         transaction,
         'data.attributes.transactionId',
-        `${fieldName}_${transaction.source_date}`,
+        `${fieldName}_${transaction?.attributes?.[fieldName]?.sourceDate}`,
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/personalInformation.js:104]: the structure of a successful PUT request is slightly different than the swagger docs, so they are being updated and the 'id' creation here also needed to be updated for the correct path to the timestamp property

File: src/applications/personalization/profile/mocks/server.js [hunk 3]
```diff
@@ -4,7 +4,11 @@ const paymentHistory = require('./paymentHistory');
 const mhvAcccount = require('./mhvAccount');
 const address = require('./address');
 const status = require('./status');
-const personalInformation = require('./personalInformation');
+const {
+  handlePutGenderIdentitiesRoute,
+  handleGetPersonalInformationRoute,
+  handlePutPreferredNameRoute,
+} = require('./personal-information');
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/mocks/server.js:11]: Moved this mock into its own folder to adhere to the other mock's organizational structure

**Review by tmitchellgcio — APPROVED** [review]
LGTM

**Review by mdewey — APPROVED** [review]
LGTM

a few linting errors, but otherwise its good to go

**Review by timwright12 — APPROVED** [review]

**Pull request merged**